cargo-public-api/src/plain.rs

```diff
@@ -49,11 +47,16 @@ impl OutputFormatter for Plain {
             &diff.changed,
             |w, changed_item| {
                 if use_color {
+                    let old_tokens: Vec<&str> =
+                        changed_item.old.tokens().map(Token::text).collect();
+                    let new_tokens: Vec<&str> =
+                        changed_item.new.tokens().map(Token::text).collect();
+                    let diff_slice = diff::slice(old_tokens.as_slice(), new_tokens.as_slice());
```

Comment:
Seems like you still mess around with strings a bit more than I would like :)

I'm thinking more along the lines of:
```suggestion
                    let diff_slice = diff::slice(
                        &changed_item.old.tokens().collect::<Vec<_>>(),
                        &changed_item.new.tokens().collect::<Vec<_>>(),
                    );
```
and then make necessary adjustments to the rest of the code from that. The above change makes the type of `diff_slice` be `Vec<Result<&&Token>>` instead of `Vec<Result<&&str>>`.

(BTW, if you happen to be interested in Rust productivity tips, I can recommend VS Code + rust-analyzer, because then you just hold down control+option to temporarily see the type of all vars etc in the code.)